src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,355 @@
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void divisionWithZero_NotOk() {
+        first = 0.0;
+        second = 0.0;
+        operation = '/';
+        assertThrows(ArithmeticException.class, () -> calculator.calculate(0.0, 0.0, operation));
+        first = 1.0;
+        second = 0.0;
+        assertThrows(ArithmeticException.class, () -> calculator.calculate(1.0, 0, operation));
+    }
+
+    @Test
+    void divisionMaxAndMinDoubleValue_NotOk() {
+        first = Double.MAX_VALUE;
+        second = 0.0;
+        operation = '/';
+        assertThrows(ArithmeticException.class,
+                () -> calculator.calculate(first, second, operation));
+    }
+
+    @Test
+    void raisingPositiveAndNegativeToPositive_Ok() {
+        first = 2.0;
+        second = -3.0;
+        expected = 0.125;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingNegativeAndPositiveToNegative_Ok() {
+        first = -2.0;
+        second = 3.0;
+        expected = -8.0;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingNegativeToNegative_Ok() {
+        first = -2.0;
+        second = -3.0;
+        expected = -0.125;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingPositiveToPositive_Ok() {
+        first = 2.0;
+        second = 3.0;
+        expected = 8.0;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingPositiveToZero_Ok() {
+        first = 2.0;
+        second = 0.0;
+        expected = 1.0;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingNegativeToZero_Ok() {
+        first = -2.0;
+        second = 0.0;
+        expected = 1.0;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingZeroToZero_Ok() {
+        first = 0.0;
+        second = 0.0;
+        expected = 1.0;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingZeroToPower_Ok() {
+        first = 0.0;
+        second = 8.0;
+        expected = 0.0;
+        operation = '^';
+        actual = calculator.calculate(first, second, operation);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void calculate_illegalOperation_notOk() {
+        first = 0.0;
+        second = 8.0;
+        assertThrows(IllegalArgumentException.class, () ->
```

Comment:
```suggestion
        second = 8.0;
        char illegalOperand = '?';
        assertThrows(IllegalArgumentException.class, () ->
```
для читаймості краще оголосити ? як змінну